toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/WorldScaleSceneView.kt

```diff
@@ -310,3 +278,210 @@ public fun WorldScaleSceneView(
         }
     }
 }
+
+/**
+ * Checks the permissions required for the [WorldScaleSceneView] to function and requests them if necessary.
+ * If [requestCameraPermissionAutomatically] or [requestLocationPermissionAutomatically] are set to false,
+ * the permissions will not be requested automatically and the initialization will fail if the permissions are not granted.
+ *
+ * @return A [MutableState] that will be true when the permissions have been requested and the result has been received.
+ * @since 200.7.0
+ */
+@Composable
+private fun requestPermissionsOrFail(
+    context: Context,
+    requestCameraPermissionAutomatically: Boolean,
+    requestLocationPermissionAutomatically: Boolean,
+    initializationStatus: MutableState<WorldScaleSceneViewStatus>,
+    onInitializationStatusChanged: ((WorldScaleSceneViewStatus) -> Unit)?
+): Boolean {
+    var hasLaunchedRequest by remember { mutableStateOf(false) }
```

Comment:
Does this variable need to be a state object? I don't think we use it to trigger recomposition.
```suggestion
    var hasLaunchedRequest = remember { false }
```

Reply:
I changed this to use `LaunchedEffect(Unit)` to launch the request. We can discuss if you think `LaunchedEffect` is appropriate here, but it permits us to use a key even though we are not suspending. The key means we can pass `Unit` and the launch only happens once, which is the behavior this boolean was meant to enable anyhow.